lib/core/src/Cardano/Wallet/Primitive/Slotting.hs

```diff
@@ -334,29 +389,52 @@ data TimeInterpreter m = forall eras. TimeInterpreter
 data TimeInterpreterLog
     = MsgInterpreterPastHorizon
         (Maybe String) -- ^ Reason for why the failure should be impossible
+        StartTime
         PastHorizonException
     deriving (Eq, Show)
 
 instance HasSeverityAnnotation TimeInterpreterLog where
     getSeverityAnnotation = \case
-        MsgInterpreterPastHorizon Nothing _ -> Notice
-        MsgInterpreterPastHorizon _ _ -> Error
+        MsgInterpreterPastHorizon Nothing _ _ -> Notice
+        MsgInterpreterPastHorizon{} -> Error
 
 instance ToText TimeInterpreterLog where
     toText = \case
-        MsgInterpreterPastHorizon Nothing e -> mconcat
+        MsgInterpreterPastHorizon Nothing t0 e -> mconcat
             [ "Time interpreter queried past the horizon. "
-            , "Full error is: "
-            , T.pack (show e)
+            , renderPastHorizonException e t0
             ]
-        MsgInterpreterPastHorizon (Just reason) e -> mconcat
+        MsgInterpreterPastHorizon (Just reason) t0 e -> mconcat
             [ "Time interpreter queried past the horizon. "
             , "This should not happen because "
             , T.pack reason
-            , " Full error is: "
-            , T.pack (show e)
+            , renderPastHorizonException e t0
+            ]
+      where
+        renderPastHorizonException (PastHorizon callStack expr eras) t0 = mconcat
+            [ "\nCalled from:\n"
+            , T.pack $ show $ prettySrcLoc $ snd $ last $ getCallStack callStack
```

Comment:
Note that `last` is partial. I hope there's always a stack :thinking:

Reply:
I changed it to not use `last`.
It would be better to pretty print the entire callstack though. Because I think this line depends on the calling function to _not_ have HasCallStack.